api/maven-api-core/src/main/java/org/apache/maven/api/Project.java

```diff
@@ -68,12 +68,31 @@ public interface Project {
     String getPackaging();
 
     /**
-     * Returns the project artifact, that is the artifact produced by this project build.
+     * Returns the project POM artifact, that is the artifact of the POM of this project. Every project have POM
+     * artifact, while the existence of backing POM file is NOT a requirement (i.e. for some transient projects).
      *
      * @see org.apache.maven.api.services.ArtifactManager#getPath(Artifact)
      */
     @Nonnull
-    Artifact getArtifact();
+    default Artifact getPomArtifact() {
+        return getArtifacts().get(0);
+    }
+
+    /**
+     * Returns the project artifact, that is the artifact produced by this project build. This artifact MAY be same
+     * as the one returned by {@link #getPomArtifact()}, if the project is actually not producing any main artifact.
+     * <p>
+     * If only non-POM artifacts are needed, better use {@link #getArtifacts()} method: if that method returns list
+     * having one element, the methods {@link #getPomArtifact()} and this one will return same artifact.
+     *
+     * @see #getArtifacts()
+     * @see org.apache.maven.api.services.ArtifactManager#getPath(Artifact)
+     */
+    @Nonnull
+    default Artifact getArtifact() {
+        List<Artifact> artifacts = getArtifacts();
+        return artifacts.get(artifacts.size() - 1);
```

Comment:
Wait a second, if there are other attached artifacts this will not work, will it?

Reply:
not true: getArtifacts always returns 1..2 artifact. getAllArtifacts considers attached ones as well...

Reply:
Right, my bad!